avocado/core/runner.py

```diff
@@ -493,15 +492,16 @@ def run_suite(self, test_suite, mux, timeout=0, replay_map=None,
         summary = set()
         if self.job.sysinfo is not None:
             self.job.sysinfo.start_job_hook()
-        self.result.start_tests()
         queue = queues.SimpleQueue()
 
         if timeout > 0:
             deadline = time.time() + timeout
         else:
             deadline = None
 
+        test_result_total = mux.get_number_of_tests(test_suite)
         no_digits = len(str(test_result_total))
+        self.result.start_tests(test_result_total)
```

Comment:
I see. So this is just a method to update the number of `total_tests` from the runner. I don't think it's necessary to do this every-time. I'd allow doing so, but as a special method (or just non-private variable) settable from runner, instead of updating this value on each `start_tests` call.

So basically it can stay as is, only we'd call `result_proxy.tests_total(xxx)` to override the default value from `RemoteResults` if necessary (this could relate to Job API, where we might want to update the number of tests from time to time...)